### PR TITLE
[VideoMAE] TensorFlow implementation

### LINKED ISSUE #18641: Add TF VideoMAE
### Feature request

Add the [VideoMAE](https://huggingface.co/docs/transformers/main/en/model_doc/videomae) model in TensorFlow.

### Motivation

There's an evident scarcity of SoTA and easy-to-use video models in TensorFlow. I believe having VideoMAE in TensorFlow would greatly benefit the community. 

### Your contribution

I am willing to contribute the model. Please assign it to me. 

@amyeroberts possible to assign this to me?

### PR DESCRIPTION
**Pull request by sayakpaul**
Closes #18641 

## TODO

- [x] modeling_tf_videomae.py
- [x] integration tests
- [x] rest of the tests
- [x] documentation

### CODE REVIEW

**Comment by sayakpaul** [comment]
I am hitting an issue and thought of asking for help.

The issue stems from the port of VideoMAESelfAttention i.e., TFVideoMAESelfAttention. Upon analyzing deeper, I think it's because of the mismatch between `nn.functional.linear` and my custom `linear_transformation()`. This is my investigative [notebook](https://colab.research.google.com/gist/sayakpaul/d50d013f59674b943ef7e2b6ed9d2f91/scratchpad.ipynb) where I have tried debugging the issue but still no luck so far. The assertion errors seem to be within low tolerance but can easily sum up.

Cc: @amyeroberts @gante @Rocketknight1

File: src/transformers/models/videomae/modeling_tf_videomae.py [hunk 1]
```diff
@@ -0,0 +1,1168 @@
...
+        self.decoder = TFVideoMAEDecoder(config, name="decoder")
+
+        self.mean = tf.convert_to_tensor(IMAGENET_DEFAULT_MEAN)[None, None, :, None, None]
+        self.std = tf.convert_to_tensor(IMAGENET_DEFAULT_STD)[None, None, :, None, None]
+
+    def build(self, input_shape: tf.TensorShape):
+        self.mask_token = self.add_weight(
+            shape=(1, 1, self.config.decoder_hidden_size), initializer="zeros", name="mask_token"
+        )
+        super().build(input_shape)
```
Comment (sayakpaul) [inline src/transformers/models/videomae/modeling_tf_videomae.py:873]: @gante @Rocketknight1 

The PT variant of this model initializes a parameter inside `PretrainedModel`:

https://github.com/huggingface/transformers/blob/8637316e5e94ba0a2493e5df7846f2f23f46eaef/src/transformers/models/videomae/modeling_videomae.py#L742

I followed it in the TF variant and it led to the obvious error:

```
ValueError: You cannot build your model by calling `build` if your layers do not support float type inputs. Instead, in order to instantiate and build your model, call your model on real tensor data (of the correct type).
```

Where do you think we should initialize it (`mask_token`) in the TF variant?

**Comment by amyeroberts** [comment]
Added WIP label to prevent PR from being closed

**Comment by sayakpaul** [comment]
No point in extending it further.

**Pull request closed without merging**